app/controllers/display_tasks_controller.rb

```diff
@@ -14,23 +10,58 @@ def new
     render :add_task_form
   end
 
+  def show
+    @task = Task.find_by id: params[:task_id]
+    render :task
+  end
+
   def create
-    @task = Task.new(create_params[:task])
+    @task = Task.create(create_params[:task])
     @task.save
 
     redirect_to :tasks
   end
 
   def destroy
-    @task = Task.find_by id: params[:task_id]
+    @task = Task.find(params[:task_id])
     @task.destroy
 
-    redirect_to "/"
+    redirect_to :tasks
+  end
+
+  def edit
+    @task = Task.find(params[:task_id])
+
+    render :edit
+  end
+
+  def update
+    @task = Task.find(params[:task_id])
+    new_name            = params[:task][:name]
+    new_description     = params[:task][:description]
+    new_person_id       = params[:task][:person_id]
+    new_completed_at    = params[:task][:completed_at]
+
+    @task.update(name:         new_name,
+                 description:  new_description,
+                 person_id:    new_person_id,
+                 completed_at: new_completed_at)
+
+    render :task
+  end
+
+  def mark_complete
+    @task = Task.find(params[:task_id])
+    new_completed_at = Time.now
+
+    @task.update(completed_at: new_completed_at)
```

Comment:
Since you're not re-using the `new_completed_at` variable, I'd recommend removing this and just adding `Time.now` right to this line